src/web/wms/src/main/java/org/geoserver/wms/web/publish/StyleTypeChoiceRenderer.java

```diff
@@ -0,0 +1,35 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wms.web.publish;
+
+import org.apache.wicket.markup.html.form.IChoiceRenderer;
+import org.geoserver.catalog.StyleType;
+
+@SuppressWarnings("serial")
+public class StyleTypeChoiceRenderer implements IChoiceRenderer {
+
+    public Object getDisplayValue(Object object) {
+        return getIdValue(object, 0);
+    }
+
+    public String getIdValue(Object object, int index) {
+        switch ((StyleType) object) {
+            case GENERIC:
+                return "Generic";
```

Comment:
It is a unique wicket id. However, I also need a display value (for which I use the same method, see `getDisplayValue` above; given that the type names are unique, this makes more sense than duplicating code). Now that the enum is its own class, moving this to the enum is definately a good idea.